test/unit/role_test.lua

```diff
@@ -0,0 +1,660 @@
+        expirationd = {
+            [task_name] = {
+                space_id = g.space.id,
+                is_expired = always_true_func_name,
+                options = options,
+            }
+        },
+    }, {is_master = true})
+
+    t.assert_equals(#cg.role.tasks(), 1)
+    t.assert_not_equals(cg.role.task(task_name), nil)
+
+    -- pass an invalid option
+    options.index = 100
+    local ok, _ = pcall(cg.role.apply_config, {
+        expirationd = {
+            [task_name] = {
+                space_id = g.space.id,
+                is_expired = always_true_func_name,
+                options = options,
+            }
+        },
+    }, {is_master = true})
+    t.assert_equals(ok, false)
+    t.assert_equals(#cg.role.tasks(), 0)
+end
+
+function g.test_apply_config_skip_is_master_only(cg)
+    local task_name = "apply_config_test"
+
+    cg.role.apply_config({
+        expirationd = {
+            [task_name] = {
+                space_id = g.space.id,
+                is_expired = always_true_func_name,
+                is_master_only = true,
+            },
+        },
+    }, {is_master = false})
+
+    t.assert_equals(#cg.role.tasks(), 0)
+end
+
+function g.test_apply_config_start_is_master_only(cg)
+    local task_name = "apply_config_test"
+
+    cg.role.apply_config({
+        expirationd = {
+            [task_name] = {
+                space_id = g.space.id,
+                is_expired = always_true_func_name,
+                is_master_only = true,
+            },
+        },
+    }, {is_master = true})
+
+    t.assert_equals(#cg.role.tasks(), 1)
+    t.assert_not_equals(cg.role.task(task_name), nil)
+end
+
+function g.test_apply_config_empty_repeat_kill_all_config_tasks(cg)
+    local local_name1 = "local_task1"
+    local local_name2 = "local_task2"
+    local config_name1 = "config_task1"
+    local config_name2 = "config_task2"
+    local config_name3 = "config_task3"
+
+    expirationd.start(local_name1, cg.space.id, function() return true end)
+    expirationd.start(local_name2, cg.space.id, function() return true end)
+
+    cg.role.apply_config({
+        expirationd = {
+            [config_name1] = {
+                space_id = g.space.id,
+                is_expired = always_true_func_name,
+            },
+            [config_name2] = {
+                space_id = g.space.id,
+                is_expired = always_true_func_name,
+            },
+            [config_name3] = {
+                space_id = g.space.id,
+                is_expired = always_true_func_name,
+            },
+        },
+    }, {is_master = false})
+
+    t.assert_equals(#cg.role.tasks(), 5)
+    t.assert_not_equals(cg.role.task(local_name1), nil)
+    t.assert_not_equals(cg.role.task(local_name2), nil)
+    t.assert_not_equals(cg.role.task(config_name1), nil)
+    t.assert_not_equals(cg.role.task(config_name2), nil)
+    t.assert_not_equals(cg.role.task(config_name3), nil)
+
+    cg.role.task(config_name3):kill()
```

Comment:
Why we need this line?

Reply:
I've added a separate test to make it more clear. See:
`test_apply_config_empty_repeat_handle_custom_stops`